src/psyclone/tests/psyir/transformations/extract_test.py

```diff
@@ -658,27 +612,50 @@ def test_extract_colouring_omp_dynamo0p3(tmpdir):
     schedule, _ = etrans.apply(child)
 
     code = str(psy.gen)
-    output = (
-        "      ! ExtractStart\n"
-        "      ! CALL write_extract_arguments(argument_list)\n"
-        "      !\n"
-        "      DO colour=1,ncolour\n"
-        "        !$omp parallel do default(shared), private(cell), "
-        "schedule(static)\n"
-        "        DO cell=1,mesh%get_last_edge_cell_per_colour(colour)\n"
-        "          !\n"
-        "          CALL ru_code(nlayers, b_proxy%data, a_proxy%data, istp, "
-        "rdt, c_proxy%data, e_proxy(1)%data, e_proxy(2)%data, "
-        "e_proxy(3)%data, ndf_w2, undf_w2, map_w2(:,cmap(colour, cell)), "
-        "basis_w2_qr, diff_basis_w2_qr, ndf_w3, undf_w3, "
-        "map_w3(:,cmap(colour, cell)), basis_w3_qr, ndf_w0, undf_w0, "
-        "map_w0(:,cmap(colour, cell)), basis_w0_qr, diff_basis_w0_qr, "
-        "np_xy_qr, np_z_qr, weights_xy_qr, weights_z_qr)\n"
-        "        END DO\n"
-        "        !$omp end parallel do\n"
-        "      END DO\n"
-        "      !\n"
-        "      ! ExtractEnd\n")
+    output = ("""
+      ! ExtractStart
+      !
+      CALL psy_data%PreStart("multikernel_invokes_7_psy", """
+              """"invoke_0:ru_code:r0", 6, 3)
+      CALL psy_data%PreDeclareVariable("a", a)
+      CALL psy_data%PreDeclareVariable("b", b)
+      CALL psy_data%PreDeclareVariable("c", c)
+      CALL psy_data%PreDeclareVariable("e", e)
+      CALL psy_data%PreDeclareVariable("istp", istp)
+      CALL psy_data%PreDeclareVariable("rdt", rdt)
+      CALL psy_data%PreDeclareVariable("b_post", b)
+      CALL psy_data%PreDeclareVariable("cell_post", cell)
+      CALL psy_data%PreDeclareVariable("colour_post", colour)
+      CALL psy_data%PreEndDeclaration
+      CALL psy_data%ProvideVariable("a", a)
+      CALL psy_data%ProvideVariable("b", b)
+      CALL psy_data%ProvideVariable("c", c)
+      CALL psy_data%ProvideVariable("e", e)
+      CALL psy_data%ProvideVariable("istp", istp)
+      CALL psy_data%ProvideVariable("rdt", rdt)
+      CALL psy_data%PreEnd
+      DO colour=1,ncolour
+        !$omp parallel do default(shared), private(cell), schedule(static)
+        DO cell=1,mesh%get_last_edge_cell_per_colour(colour)
+          !
+          CALL ru_code(nlayers, b_proxy%data, a_proxy%data, istp, rdt, """
+              "c_proxy%data, e_proxy(1)%data, e_proxy(2)%data, "
+              "e_proxy(3)%data, ndf_w2, undf_w2, "
+              "map_w2(:,cmap(colour, cell)), "
+              "basis_w2_qr, diff_basis_w2_qr, ndf_w3, undf_w3, "
+              "map_w3(:,cmap(colour, cell)), basis_w3_qr, ndf_w0, undf_w0, "
+              "map_w0(:,cmap(colour, cell)), basis_w0_qr, diff_basis_w0_qr, "
+              """np_xy_qr, np_z_qr, weights_xy_qr, weights_z_qr)
+        END DO
+        !$omp end parallel do
+      END DO
+      CALL psy_data%PostStart
+      CALL psy_data%ProvideVariable("b_post", b)
+      CALL psy_data%ProvideVariable("cell_post", cell)
```

Comment:
x-fail please as e.g. maps and weights are not written out.

Reply:
Ditto.